src/software/simulated_tests/simulated_er_force_sim_tactic_test_fixture.cpp

```diff
@@ -0,0 +1,188 @@
+    }
+}
+
+void SimulatedErForceSimTacticTestFixture::setFriendlyRobotId(RobotId friendly_robot_id)
+{
+    this->friendly_robot_id = friendly_robot_id;
+}
+
+void SimulatedErForceSimTacticTestFixture::setBothRobotId(RobotId friendly_robot_id,
+                                                          RobotId enemy_robot_id)
+{
+    this->friendly_robot_id = friendly_robot_id;
+    this->enemy_robot_id    = enemy_robot_id;
+}
+
+void SimulatedErForceSimTacticTestFixture::setMotionConstraints(
+    const std::set<MotionConstraint>& friendly_motion_constraints,
+    const std::set<MotionConstraint>& enemy_motion_constraints)
+{
+    this->friendly_motion_constraints = friendly_motion_constraints;
+    this->enemy_motion_constraints    = enemy_motion_constraints;
+}
+
+void SimulatedErForceSimTacticTestFixture::updatePrimitives(
+    const World& friendly_world, const World& enemy_world,
+    std::shared_ptr<ErForceSimulator> simulator_to_update)
+{
+    updateFriendlyPrimitives(friendly_world, simulator_to_update);
+    if (this->enemy_robot_id)
+    {
+        updateEnemyPrimitives(enemy_world, simulator_to_update);
+    }
+}
+
+void SimulatedErForceSimTacticTestFixture::updateFriendlyPrimitives(
+    const World& world, std::shared_ptr<ErForceSimulator> simulator_to_update)
+{
+    std::vector<std::unique_ptr<Intent>> intents;
+    auto start_tick_time = std::chrono::system_clock::now();
+
+    if (!friendly_robot_id)
+    {
+        LOG(FATAL) << "No friendly robot id set" << std::endl;
+    }
+    else if (auto new_robot = world.friendlyTeam().getRobotById(*friendly_robot_id))
+    {
+        auto intent = friendly_tactic->get(
+            *world.friendlyTeam().getRobotById(*friendly_robot_id), world);
+        intent->setMotionConstraints(friendly_motion_constraints);
+        intents.push_back(std::move(intent));
+    }
+    else
+    {
+        LOG(FATAL) << "No friendly robot with robot id " << *friendly_robot_id
+                   << std::endl;
+    }
+
+    auto primitive_set_msg = friendly_navigator->getAssignedPrimitives(world, intents);
+    double duration_ms     = ::TestUtil::millisecondsSince(start_tick_time);
+    registerFriendlyTickTime(duration_ms);
+    auto vision_msg = createVision(world);
+    simulator_to_update->setYellowRobotPrimitiveSet(*primitive_set_msg,
+                                                    std::move(vision_msg));
+}
+
+void SimulatedErForceSimTacticTestFixture::updateEnemyPrimitives(
+    const World& world, std::shared_ptr<ErForceSimulator> simulator_to_update)
+{
+    std::vector<std::unique_ptr<Intent>> intents;
+    auto start_tick_time = std::chrono::system_clock::now();
+
+    if (!enemy_robot_id)
+    {
+        LOG(FATAL) << "No enemy robot id set" << std::endl;
+    }
+    else if (auto new_robot = world.friendlyTeam().getRobotById(*enemy_robot_id))
+    {
+        auto intent =
+            enemy_tactic->get(*world.friendlyTeam().getRobotById(*enemy_robot_id), world);
```

Comment:
should this be world.enemyTeam()?

Reply:
good point

Reply:
turns out no, because this is the enemy_world